src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -1065,7 +1079,7 @@ MatterThermostatClusterServerPreAttributeChangedCallback(const app::ConcreteAttr
             if (requested > OccupiedCoolingSetpoint - DeadBandTemp)
                 return imcode::InvalidValue;
         }
-        return imcode::Success;
+        break;
```

Comment:
All the changed to this function seem unrelated to presets and should be done in a separate PR, if at all.